app/views/datagrid/_dynamic_filter.html.erb

```diff
@@ -0,0 +1,19 @@
+<%# locals: (filter:, form:, field_options:, operation_options:, value_options:) -%>
+
+<%= form.datagrid_filter_input(
+      filter,
+      class: [*field_options.delete(:class), "datagrid-dynamic-field"],
+      **field_options
```

Comment:
I think it will be a little better memory footprint and a little better clarity as we avoid modifying `field_options`.

```suggestion
      **field_options
      class: [*field_options[:class], "datagrid-dynamic-field"],
```